openstack_controller/datadog_checks/openstack_controller/retry_backoff.py

```diff
@@ -0,0 +1,55 @@
+# (C) Datadog, Inc. 2010-2017
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+import time
+import random
+
+from .utils import get_instance_key
+
+BASE_BACKOFF_SECS = 15
+MAX_BACKOFF_SECS = 300
+
+
+class BackOffRetry(object):
+
+    def __init__(self, check):
+        self.backoff = {}
+        random.seed()
+        self.check = check
```

Comment:
This feels odd. What do you think about storing just the gauge function?

Reply:
Good point. However, I am not going to do it here since this PR is mainly to split the main python file into multiple. We want to change the logic of this retry logic anyway. So I will keep this comment in mind for when we will work on it.